src/Chapter3.hs

```diff
@@ -480,7 +532,81 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
-{-
+data Castle
+    = Castle String
+    | NoCastle
+
+data Wall
+    = Wall
+    | NoWall
+
+-- tried to give both Wall and Castle a None alternative but the compilare complained
+-- about multiple declarations of None, is there any way to bypass this ?
+
+data Center
+    = Church
+    | Library
+
+data House = MkHouse
+    { people  :: Int }
```

Comment:
That is a nice solution, however, it allows having any amount of people in one house, however, task specifies only a few cases of possible outcomes. You probably need another sum type to represent this 🙂